xbmc/cores/RetroPlayer/guicontrols/GUIGameControl.h

```diff
@@ -19,44 +19,64 @@
  */
 #pragma once
 
-#include "cores/RetroPlayer/rendering/GUIRenderSettings.h"
 #include "guilib/GUIControl.h"
 #include "guilib/GUIInfoTypes.h"
 
+#include <memory>
+
 class CGUIInfoLabel;
 
 namespace KODI
 {
 namespace RETRO
 {
+class CGUIRenderSettings;
+class CGUIRenderHandle;
+class IGUIRenderSettings;
 
 class CGUIGameControl : public CGUIControl
 {
 public:
   CGUIGameControl(int parentID, int controlID, float posX, float posY, float width, float height);
-  ~CGUIGameControl() override = default;
+  CGUIGameControl(const CGUIGameControl &other);
+  ~CGUIGameControl() override;
 
-  void SetVideoFilter(const CGUIInfoLabel &videoFilter);
+  // GUI functions
+  void SetScalingMethod(const CGUIInfoLabel &scalingMethod);
   void SetViewMode(const CGUIInfoLabel &viewMode);
 
-  const CGUIRenderSettings &GetRenderSettings() const { return m_renderSettings; }
+  // Rendering functions
+  bool HasScalingMethod() const { return m_bHasScalingMethod; }
+  bool HasViewMode() const { return m_bHasViewMode; }
+  IGUIRenderSettings *GetRenderSettings() const;
 
   // implementation of CGUIControl
   CGUIGameControl *Clone() const override { return new CGUIGameControl(*this); };
   void Process(unsigned int currentTime, CDirtyRegionList &dirtyregions) override;
   void Render() override;
   void RenderEx() override;
   bool CanFocus() const override;
+  void SetPosition(float posX, float posY) override;
+  void SetWidth(float width) override;
+  void SetHeight(float height) override;
   void UpdateInfo(const CGUIListItem *item = nullptr) override;
 
 private:
-  void EnableGUIRender();
-  void DisableGUIRender();
+  void Reset();
+
+  void RegisterControl();
+  void UnregisterControl();
 
-  CGUIInfoLabel m_videoFilterInfo;
+  // GUI properties
+  CGUIInfoLabel m_scalingMethodInfo;
   CGUIInfoLabel m_viewModeInfo;
 
-  CGUIRenderSettings m_renderSettings;
+  // Rendering properties
+  bool m_bHasShaderPreset = false;
```

Comment:
unused private field

Reply:
Good catch, this snuck its way in from https://github.com/garbear/xbmc/pull/86